### PR TITLE
Feature/authors info display

### PR DESCRIPTION
**Pull request by Sathya-S3**
Major updates:

1. The individual results page for a record pings crossref API to retrieve the authors of the dataset if a publication DOI is present in the record metadata.
2. The author names link to their ORCID profiles if the ORCID number is returned by crossref.

Additional updates:

1. Calculation details box is made expandable through mouse click rather than by hovering. The metadata can be copied using a 'Copy calculation details' button at the bottom of the calculation metadata window, which is cleaner than mouse highlight to select text and then copy.
2. The calculation details display is cleaned up to display a neater JSON parsed text.
3. Cleaned up table display to remove a strange gap.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -63,6 +64,12 @@ def search():
     return serv.search()
 
 
+@app.route('/api/works', methods=['GET'])
```
Comment (jkshenton) [inline main.py:67]: I find this route name a little confusing

File: static/js/record.js [hunk 2]
```diff
@@ -166,6 +185,40 @@ function addRecordDirective(ngApp) {
...
+
+                scope.isExpanded = false;
+
+                // Function to fetch author information
+                function fetchAuthorInfo(doi) {
+                    if (doi) {
+                        DoiAuthorsService.getAuthorInfo(doi).then(function(authorsList) {
+                            scope.authorsList = authorsList;
+                        });
+                    } else {
```
Comment (jkshenton) [inline static/js/record.js:197]: Without a live copy it's a bit hard to test out the functionality here, but if the record does _not_ have a DOI, does it just hide this whole section, or is the "Author information loading..." text displayed?

File: static/templates/database_record.html [hunk 3]
```diff
@@ -77,20 +74,31 @@
                 <th class="width-1">DOI:</th>
                 <td class="width-3" colspan="3"><a href="https://doi.org/{{databaseRecord.version_history[_selected_index].doi}}" target="_blank">{{databaseRecord.version_history[_selected_index].doi}}</a></td>
             </tr>
+            <tr ng-if="databaseRecord.version_history[_selected_index].doi">
```
Comment (jkshenton) [inline static/templates/database_record.html:77]: Ah ok, so here is where the authors are displayed or not depending on if the DOI is present?

**Review by jkshenton — COMMENTED** [review]
Thanks Sathya, those are very useful changes!

A quick question about the crossref lookup: does the loading of the record page wait until the lookup is complete? If so, what happens if the lookup hangs? Or does it update that content once it received the info from crossref? 

I would suggest three small changes: 

1. the naming of the `api/works` route - is there a  reason for this route name that I'm missing?
2. A few tests for the crossref lookup - add in a few random DOIs with author lists you look up and test to see that they're correct.
3. Handling of 'bad' DOIs: Since users can enter ~anything into our DOI field, we should handle the case where the crossref lookup comes back empty/with an error of some kind. Having a bad DOI in the tests would also be useful. 

Cheers

File: main.py (hunk 1 above)
Comment (Sathya-S3) [reply main.py:67]: I suppose something simpler like get_authors can work just as well. I'll test this and confirm.

File: static/templates/database_record.html (hunk 3 above)
Comment (Sathya-S3) [reply static/templates/database_record.html:77]: Yes, that's right.

**Comment by Sathya-S3** [comment]
> Thanks Sathya, those are very useful changes!
> 
> A quick question about the crossref lookup: does the loading of the record page wait until the lookup is complete? If so, what happens if the lookup hangs? Or does it update that content once it received the info from crossref?
> 
> I would suggest three small changes:
> 
> 1. the naming of the `api/works` route - is there a  reason for this route name that I'm missing?
> 2. A few tests for the crossref lookup - add in a few random DOIs with author lists you look up and test to see that they're correct.
> 3. Handling of 'bad' DOIs: Since users can enter ~anything into our DOI field, we should handle the case where the crossref lookup comes back empty/with an error of some kind. Having a bad DOI in the tests would also be useful.
> 
> Cheers

@jkshenton Besides making changes as you suggested, I have added test cases to fetch 1 good doi and 1 bad doi.

For good doi, I have pre-defined authors' names as list of strings to compare that the names are in the returned response (author names part of the total API response). For bad doi, the test checks that he appropriate error message and the error status code are returned. Please see commit [de3872a](https://github.com/CCP-NC/ccpnc-database/pull/43/commits/de3872a12b87af9382951ffd10eee9b9d6a626eb)

Do you have any improvement suggestions?

**Comment by jkshenton** [comment]
Hi @Sathya-S3 , that looks good to me now - thanks!

**Pull request merged**